lib/permissions.js

```diff
@@ -17,42 +23,50 @@ const SERVICES = {
   speech: 'kTCCServiceSpeechRecognition',
 };
 
-const STATUS_UNSET = 'unset';
-const STATUS_YES = 'yes';
-const STATUS_NO = 'no';
-
-
 function toInternalServiceName (serviceName) {
   if (_.has(SERVICES, _.toLower(serviceName))) {
     return SERVICES[_.toLower(serviceName)];
   }
   throw new Error(`'${serviceName}' is unknown. Only the following service names are supported: ${JSON.stringify(_.keys(SERVICES))}`);
 }
 
-function assertStatus (status) {
-  const supportedValues = [STATUS_UNSET, STATUS_NO, STATUS_YES];
-  if (!supportedValues.includes(status)) {
-    throw new Error(`Status value '${status}' is unknown. Only the following values are supported: ${JSON.stringify(supportedValues)}`);
+function formatStatus (status) {
+  return [STATUS_UNSET, STATUS_NO].includes(status) ? _.toUpper(status) : status;
+}
+
+async function execWix (args) {
+  try {
+    await fs.which(WIX);
+  } catch (e) {
+    throw new Error(`${WIX} binary has not been found in your PATH. ` +
+      `Please install it ('brew tap wix/brew && brew install wix/brew/applesimutils') to ` +
+      `be able to change application permissions`);
+  }
+
+  log.debug(`Executing ${WIX} with arguments ${JSON.stringify(args)}`);
+  try {
+    return (await exec(WIX, args)).stdout;
+  } catch (e) {
+    throw new Error(`Cannot execute ${WIX} with arguments ${JSON.stringify(args)}. ` +
+      `Original error: ${e.stderr || e.message}`);
   }
 }
 
 class Permissions {
-
-  constructor (xcodeVersion, sharedResourcesDir) {
+  constructor (xcodeVersion, sharedResourcesDir, udid) {
     this.tccDb = new TCCDB(xcodeVersion, sharedResourcesDir);
+    this.udid = udid;
   }
 
-  async setAccess (serviceName, bundleId, status) {
-    serviceName = toInternalServiceName(serviceName);
-    assertStatus(status);
-
-    await retryInterval(15, 2000, async () => {
-      await this.tccDb.execQuery(`DELETE FROM 'access' WHERE service='?' AND client='?' AND client_type=?`, serviceName, bundleId, '0');
-      if (status !== STATUS_UNSET) {
-        await this.tccDb.execQuery(`REPLACE INTO 'access' (service, client, client_type, allowed, prompt_count) VALUES ('?', '?', ?, ?, ?)`,
-          serviceName, bundleId, '0', status === STATUS_YES ? '1' : '0', '1');
-      }
-    });
+  async setAccess (permissionsMapping, bundleId) {
+    const permissionsArg = _.toPairs(permissionsMapping)
```

Comment:
What is the expected "permissionsMapping"?

Reply:
Will add doctring